library/core/src/intrinsics/simd.rs

```diff
@@ -0,0 +1,420 @@
+    ///
+    /// `T` must be a vector of integer or floating point primitive types.
+    ///
+    /// # Safety
+    /// For integers, `rhs` must not contain any zero elements.
+    /// Additionally for signed integers, `<int>::MIN / -1` is undefined behavior.
+    pub fn simd_div<T>(lhs: T, rhs: T) -> T;
+
+    /// Remainder of two vectors elementwise
+    ///
+    /// `T` must be a vector of integer or floating point primitive types.
+    ///
+    /// # Safety
+    /// For integers, `rhs` must not contain any zero elements.
+    /// Additionally for signed integers, `<int>::MIN / -1` is undefined behavior.
+    pub fn simd_rem<T>(lhs: T, rhs: T) -> T;
+
+    /// Elementwise vector left shift, with UB on overflow.
+    ///
+    /// Shift `lhs` left by `rhs`, shifting in sign bits for signed types.
+    ///
+    /// `T` must be a vector of integer primitive types.
+    ///
+    /// # Safety
+    ///
+    /// Each element of `rhs` must be less than `<int>::BITS`.
+    pub fn simd_shl<T>(lhs: T, rhs: T) -> T;
+
+    /// Elementwise vector right shift, with UB on overflow.
+    ///
+    /// `T` must be a vector of integer primitive types.
+    ///
+    /// Shift `lhs` right by `rhs`, shifting in sign bits for signed types.
+    ///
+    /// # Safety
+    ///
+    /// Each element of `rhs` must be less than `<int>::BITS`.
+    pub fn simd_shr<T>(lhs: T, rhs: T) -> T;
+
+    /// Elementwise vector "and".
+    ///
+    /// `T` must be a vector of integer primitive types.
+    pub fn simd_and<T>(x: T, y: T) -> T;
+
+    /// Elementwise vector "or".
+    ///
+    /// `T` must be a vector of integer primitive types.
+    pub fn simd_or<T>(x: T, y: T) -> T;
+
+    /// Elementwise vector "exclusive or".
+    ///
+    /// `T` must be a vector of integer primitive types.
+    pub fn simd_xor<T>(x: T, y: T) -> T;
+
+    /// Numerically cast a vector, elementwise.
+    ///
+    /// `T` and `U` must be vectors of integer or floating point primitive types, and must have the
+    /// same length.
+    ///
+    /// When casting floats to integers, the result is truncated. Out-of-bounds result lead to UB.
+    /// When casting integers to floats, the result is rounded.
+    /// Otherwise, truncates or extends the value, maintaining the sign for signed integers.
+    ///
+    /// # Safety
+    /// Casting floats to integers truncates, but the truncated value must fit in the target type.
+    pub fn simd_cast<T, U>(x: T) -> U;
+
+    /// Numerically cast a vector, elementwise.
+    ///
+    /// `T` and `U` be a vectors of integer or floating point primitive types, and must have the
+    /// same length.
+    ///
+    /// Like `simd_cast`, but saturates float-to-integer conversions.
+    /// This matches regular `as` and is always safe.
+    ///
+    /// When casting floats to integers, the result is truncated.
+    /// When casting integers to floats, the result is rounded.
+    /// Otherwise, truncates or extends the value, maintaining the sign for signed integers.
```

Comment:
What does NaN turn into? 0 or MIN or MAX?

Reply:
0, like `as`. I added it to the docs